### ISSUE TITLE
Implement GBIF IPT

### ISSUE THREAD

**Issue by DavidRoy** [BiologicalRecordsCentre/MammalNet-Europe#5]
To enable automated data flow from Indicia to GBIF

**Comment by DavidRoy** [comment]
@johnvanbreda could you give an estimate of the time needed to implement a GBIF IPT and start supplying data.  Also, when you could schedule this work.  So I can update the project partners

**Comment by johnvanbreda** [comment]
I can prioritise this work between now and Christmas if that suits? Before starting, I'd want Biren to set up a server with an installation of the IPT on it for me to configure. I'd previously estimated up to 3 days for me to get the configuration right.

**Comment by DavidRoy** [comment]
Please go ahead with this as discussed by email.  @BirenRathod to support with server setup.

**Comment by BirenRathod** [comment]
@johnvanbreda 
Send me the details about server requirements, so I can start looking into.

**Comment by johnvanbreda** [comment]
@BirenRathod the details on installation and server requirements are at https://github.com/gbif/ipt/wiki/IPT2ManualNotes.wiki#getting-started-guide
Thanks.

**Comment by BirenRathod** [comment]
@johnvanbreda 
Thanks for the link.
Do you want me setup first on dev server or shall i just get the new server?

**Comment by johnvanbreda** [comment]
I don't mind, if you think the devserver is capable that is fine. When we go live though we probably want it to be separate to warehouse1 and the main Drupal web server.

**Comment by BirenRathod** [comment]
@johnvanbreda 
Ok, thanks for your view. I will investigate and decide from there.

**Comment by johnvanbreda** [comment]
Hi @BirenRathod let me know if you need any more info from me.

**Comment by BirenRathod** [comment]
@johnvanbreda 
I'm still waiting for response from our guy in helpdesk team to provide me another VM. I checked yesterday but the guy who deal with VM, is not feeling well. As soon as, I get VM, I'll soon start setting set up GBIF IPT.

**Comment by BirenRathod** [comment]
@johnvanbreda & @DavidRoy 

The server has setup and to access console please use this URL -> http://192.171.182.54:8080/ipt .

@johnvanbreda you need to log in first to VPN to access the URL. Same as elastic one. Although you don't need an access of the server but if you need let me know too.

**Comment by BirenRathod** [comment]
@johnvanbreda 
Further to this...
I have created you user account. I sent you text of password.

**Comment by johnvanbreda** [comment]
@BirenRathod did you set up the IPT in test mode or production mode? 

Also, I think it will be necessary to make this IP address (or a suitable domain link to the server) available from outside the firewall so we can link it to the GBIF servers, even for testing purposes.

Thanks.

**Comment by BirenRathod** [comment]
@johnvanbreda 
It is in prodcution mode. I haven't ticked the check box for debugging mode. There is no test mode.

**Comment by BirenRathod** [comment]
@johnvanbreda 
About domain, I'm thinking something like gbif.brc.ac.uk. if that's ok, I will ask our domain support team to add one.

**Comment by johnvanbreda** [comment]
The domain sounds fine to me - or ipt.brc.ac.uk would be OK as well. About test mode, you only get an option to turn this on during the install according to the instructions. I think it is important we start with a server in test mode, then re-install as a production mode server only when we are happy with how to set it up.

**Comment by johnvanbreda** [comment]
@BirenRathod please can you let me know when you have a server set up in test mode so I can proceed with this. Thanks.

**Comment by BirenRathod** [comment]
@johnvanbreda 
I know have emailed you regarding this but to update the status of this issue I need to add URL here too.
It is ipt.brc.ac.uk/ipt. Also I have already setup your user account and sent you a password too.

**Comment by DavidRoy** [comment]
@johnvanbreda @BirenRathod where are we with the setup of the GBIF IPT given it spans a number or projects

**Comment by BirenRathod** [comment]
@johnvanbreda 
It is now with John. He can start doing work on it and this is the URL -> http://ipt.brc.ac.uk/ipt

**Comment by johnvanbreda** [comment]
@BirenRathod thanks for setting this up. Do I have admin rights, or restricted rights on the IPT? I've tried to register a dataset and cannot add a "publishing organisation" so I wonder if the ability to add an organisation is an admin thing?

**Comment by BirenRathod** [comment]
@johnvanbreda 
You now got a admin rights, so you can do same as admin.

**Comment by johnvanbreda** [comment]
@DavidRoy please can you arrange for the EML document to be filled in on the following auto-export definition:
https://www.brc.ac.uk/irecord/node/7976/edit

**Comment by DavidRoy** [comment]
@johnvanbreda I've drafted some text.

**Comment by johnvanbreda** [comment]
Thanks @DavidRoy. We need the EML in XML format in the EML field on the edit page of the export, so I copied the vascular plants export's one and edited it with the text you provided. If we are providing the same set of fields then we still need to fill in:
* dataset title
* check licence is correct
* I set GeographicDescription to Europe?
* Creator email address
* GBIF citation

**Comment by DavidRoy** [comment]
@johnvanbreda The metadata is now edited.  What's the next step?  Is it possible to restrict to verified records?

**Comment by johnvanbreda** [comment]
@DavidRoy Ok, now the EML is filled in it turns out the IPT doesn't read it and it has to be filled in manually :(. I can copy this information over but will also need to provide contact details for a resource creator and a resource contact (which can be the same). Any preferences who these should point to?

**Comment by DavidRoy** [comment]
Resource creator should be Biological Records Centre and email: brc@ceh.ac.uk

**Comment by johnvanbreda** [comment]
It needs an associated person name unfortunately.

**Comment by DavidRoy** [comment]
ok, me in that case.  David Roy, dbr@ceh.ac.uk

**Comment by johnvanbreda** [comment]
@DavidRoy Would it be OK to ask you to log into the IPT, go to the Manage Resources tab, edit the dataset, edit it's metadata, then go through all the metadata categories on the right and fill them in? I started doing some but realised it's just going to generate questions for you (e.g. temporal coverage).

**Comment by DavidRoy** [comment]
That is a lot of stuff.  Hopefully now all complete

**Comment by johnvanbreda** [comment]
@DavidRoy. Thanks. I've published the dataset but not registered it with GBIF. Shall I proceed? I'm just slightly cautious as I can't see what the dataset will look like on GBIF until it's actually done.

**Comment by DavidRoy** [comment]
@johnvanbreda I'm not sure how else we'd test so this so please go ahead.  Can you confirm this is just verified records to start with

**Comment by johnvanbreda** [comment]
Done and yes - checked that we are publishing just accepted records.
See https://www.gbif.org/dataset/3ccfe992-043a-4fff-951d-1e6e158854da

**Comment by DavidRoy** [comment]
Thanks.  Looks good.  Can you remind me of the frequency of updates?

**Comment by johnvanbreda** [comment]
The process is manual at the moment:
1) Use the iRecord Exports system to generate a new Darwin Core archive.
2) Use the IPT to add the file to the existing resource, overwriting the old files.
3) Check the mappings (should be OK but I had a problem when I did this which I think was me, but just to be sure).
4) Publish the new version using the IPT.

I suspect that we could implement a clever solution which runs on the IPT server to re-build the source file and place it in the IPT data directory on a schedule. The IPT can be configured to periodically publish. This would be a few days work I suspect.

**Comment by DavidRoy** [comment]
Thanks. I'll close this issue now and will add a separate issue under another project to cover the time for an automated solution.
